crates/issue/src/lib.rs

```diff
@@ -400,7 +408,17 @@ impl<T: Config> Pallet<T> {
                     issue.griefing_collateral()
                 };
 
-                to_release_griefing_collateral.unlock_on(&requester)?;
```

Comment:
I'm not sure we should change the `execute_issue` code. If we really want to prevent persistent self-issues, then we should implement an off-chain worker to cancel expired issues automatically. And if we have that, we no longer need this change in `execute_issue`. It'd also be easier for e.g. squid to keep track of confiscated griefing collateral that way (plus it keeps the code simpler with fewer edge cases)